### ISSUE TITLE
Fix tasks running plot so that it gives a consistent view

### ISSUE THREAD

**Issue by klannon** [matz-e/lobster#367]
It makes no sense for "all" to be lower than the sum of the running tasks.  I understand that there are reasons for this feature, but for the non-expert user this is really confusing.

![tasks-plot](https://cloud.githubusercontent.com/assets/5149849/15293608/c55130aa-1b56-11e6-935d-2d51a11e0dfa.png)

Proposed solution:  Make "all" an explicit sum of the components.  If there is any information in the difference between the way "all" and the category results are obtained, this should be extracted and presented to the user is a different plot and a more intuitive way.

**Comment by matz-e** [comment]
FWIW, we currently put all merge tasks into a 'merge' category, which is missing from that plot. We should add that, too (this requires saving its info in the first place).

**Comment by matz-e** [comment]
@btovar: `self.queue.stats_hierarchy` is less than the sum of `self.queue.stats_category(category)` for almost all categories… Is this a bug in WQ or can we just not compare the two?

**Comment by klannon** [comment]
FYI: This was discussed here:
https://ndlobster.slack.com/archives/lobster-scale-tests/p1463228592000466

Excerpt:
```
btovar [11:18 AM]
I think I figure it out. "total" gives up to date from what the workers think is running.
[11:18] 
stats_category are all the tasks the master think are running (even if they are already completed at a worker)

annawoodard [11:19 AM] 
So the difference is all the tasks that are waiting?

btovar [11:19 AM] 
If the master is overwhelmed, those two may get out of sync.
[11:19] 
waiting and completed waiting to be retrieved.
[11:21] 
total and per category do measure different things. When the master is not overwhelmed, they mostly match.
[11:21] 
For your plot, perhaps you want to manually add per category?
```

**Comment by matz-e** [comment]
Argh… thanks! I clearly have too many things I follow and try to work on… This means we can actually show a plot with the "lag" of the master compared to the workers and give a measure of how well the system is handling the load?

**Comment by btovar** [comment]
@matz-e: This is a known issue. Running tasks in stats_category should be really named "dispatched by the master". stats_hierarchy is closer to tasks actually executing (e.g., not waiting in a worker to be executed, or completed to be returned).

This made sense for the foremen, as tasks waiting at a foreman were shown in the graphs as running, which was confusing. Doing this per category takes a little of more work, and I am thinking of clean ways of doing it. 

Further, if you were to use foremen, stats_category would paint a misleading picture of all dispatched tasks actually running.

**Comment by btovar** [comment]
(Yes, that, what @klannon said :) )

**Comment by btovar** [comment]
Yes, in this case the difference is a very good indicative of tasks completed waiting to be returned, as you are not using over-commitment and tasks should start executing as soon as they land on a worker.

**Comment by matz-e** [comment]
I think we should actually add the 'merge' category to what we plot — then we should have all information to be able to sum it up. I think 'dispatched' vs 'running' could then hint at the need to use foremen or find bottlenecks.

**Comment by matz-e** [comment]
I'll see if I have time to work on that today…

**Comment by btovar** [comment]
That's my thinking too. Separate running and dispatched. It should make easier the accounting at the foreman too, as now we are trying to shoehorn running and dispatched under the same statistic.

**Comment by matz-e** [comment]
If we could get that disentangled, that would be great :)
